presto-spi/src/main/java/com/facebook/presto/spi/relation/DeterminismEvaluator.java

```diff
@@ -0,0 +1,19 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.relation;
+
+public interface DeterminismEvaluator
```

Comment:
Can we also rename `com.facebook.presto.sql.planner.DeterminismEvaluator` into `ExpressionDeterminismEvaluator` and mark it `deprecated`? That class is a complete hack. It is also incorrect to some extend. It should go away in the future.

Reply:
Will do.